### DISCUSSION TITLE
Understanding the Order of Event Triggers in Mongock and Spring

### DISCUSSION THREAD

**Discussion by ahmadsab95** [flamingock/mongock#23] [Q&A] [answered]
Could you please provide an explanation of the order in which events are triggered by Mongock in relation to Spring events? For example, I'm aware of the Spring event called 'ApplicationReadyEvent.' Could you clarify whether this event occurs before or after the 'SpringMigrationSuccessEvent' provided by Mongock?

**Comment by osantana85** [comment, marked as answer]
Hello @ahmadsab95 , The `SpringMigrationSuccessEvent` / `SpringMigrationFailureEvent` events are triggered at the end of the Mongock's migration.  As Mongock's execution happens within application startup (and before Springboot finish its startup), the Mongock's events will be triggered before Springboot's `ApplicationReadyEvent` event.

Note that Springboot is the one responsible of guaranteeing this, you just need to read the documentation about ApplicationRunner and InitializingBean and use the one that better fits your requirements. 

For further details about Mongock's events please take a look to the [Springboot runner events section](url) in out documentation website.

Hope this helps.
